### PR TITLE
fix: Fix bug when Magento MSI is not enabled

### PR DESCRIPTION
**Pull request by MatthijsBreed**
When Magento MSI is not enabled, the `bin/magento setup:di:command` fails on `Model/Source/SourceItem.php`
This is because the objectmanager cannot instantiate the `Magento\InventoryApi\Api\SourceItemRepositoryInterface`
```
Area configuration aggregation... 5/9 [===============>------------]  55% 18 secs 334.0 MiB
In ClassReader.php line 45:
                                                                                                                                                                                                           
  Impossible to process constructor argument Parameter #1 [ <required> Magento\InventoryApi\Api\SourceItemRepositoryInterface $sourceItemRepository ] of MyParcelNL\Magento\Model\Source\SourceItem class                                                                                                              

In ClassReader.php line 34:
                                                                               
  Class Magento\InventoryApi\Api\SourceItemRepositoryInterface does not exist  
                                                                               
111
This PR applies the same fix as PR #577 to the SourceItem class itself

### CODE REVIEW

File: Model/Source/SourceItem.php [hunk 1]
```diff
@@ -13,31 +14,56 @@ class SourceItem
...
 
-        return $this->sourceItemRepository->getList($searchCriteria)->getItems();
+    /**
+     * Check if the module Magento_InventoryApi is activated.
+     * Some customers have removed the Magento_InventoryApi from their system.
+     * That causes problems with the Multi Stock Inventory
+     *
+     * @return void
+     */
+    private function setSourceItemRepositoryWhenInventoryApiEnabled()
```
Comment (RichardPerdaan) [inline Model/Source/SourceItem.php:63]:
```suggestion
    private function setSourceItemRepositoryWhenInventoryApiEnabled(): void
```

File: Model/Source/SourceItem.php [hunk 2]
```diff
@@ -13,31 +14,56 @@ class SourceItem
...
-            ->create();
+        if ($this->sourceItemRepository) {
+            $searchCriteria = $this->searchCriteriaBuilder
+                ->addFilter(SourceItemInterface::SKU, $sku)
+                ->create();
+
+            return $this->sourceItemRepository->getList($searchCriteria)->getItems();
+        } else {
+            return [];
+        }
```
Comment (RichardPerdaan) [inline Model/Source/SourceItem.php:52]: This else is not necessary, what you could do is replace this:
```
} else {
    return [];
}
```
 
To:
```
}   

return [];
```

**Review by RichardPerdaan — CHANGES_REQUESTED** [review]

File: Model/Source/SourceItem.php [hunk 3]
```diff
@@ -13,31 +14,56 @@ class SourceItem
...
+    /**
+     * @var Manager
+     */
+    private $moduleManager;
+
     /**
      * @var SourceItemRepositoryInterface
      */
-    private $sourceItemRepository;
+    private $sourceItemRepository = null;
```
Comment (EdieLemoine) [inline Model/Source/SourceItem.php:25]: You don't have to define null explicitly
```suggestion
    private $sourceItemRepository;
```

**Review by EdieLemoine — CHANGES_REQUESTED** [review]

File: Model/Source/SourceItem.php [hunk 4]
```diff
@@ -13,31 +14,56 @@ class SourceItem
...
 
     public function __construct(
         SearchCriteriaBuilder $searchCriteriaBuilder,
-        SourceItemRepositoryInterface $sourceItemRepository
+        Manager $moduleManager
     ) {
         $this->searchCriteriaBuilder = $searchCriteriaBuilder;
-        $this->sourceItemRepository = $sourceItemRepository;
+        $this->moduleManager = $moduleManager;
```
Comment (RichardPerdaan) [inline Model/Source/SourceItem.php:32]: `$this->moduleManager` should be formatted like
```  
$this->searchCriteriaBuilder = $searchCriteriaBuilder;
$this->moduleManager         = $moduleManager;

**Review by RichardPerdaan — CHANGES_REQUESTED** [review]

**Comment by RichardPerdaan** [comment]
Hi @MatthijsBreed ,
Thank you for your pull request, I appreciate that. 

We found some small feedback points. 
Once you've changed it, then it will be included in the next release.

File: Model/Source/SourceItem.php (hunk 3 above)
Comment (MatthijsBreed) [reply Model/Source/SourceItem.php:25]: I followed how the SourceItem was declared in `Model\Sales\MagentoOrderCollection` for consistency:
```
class MagentoOrderCollection extends MagentoCollection
{
    /**
     * @var \Magento\Sales\Model\ResourceModel\Order\Collection
     */
    private $orders = null;

    /**
     * @var SourceItem
     */
    private $sourceItem = null;
```
I can remove it if so desired

**Comment by EgorDm** [comment]
Hi, will this be merged soon or are there still changes needed?
Thanks

@MatthijsBreed @RichardPerdaan

**Review by EdieLemoine — APPROVED** [review]

**Review by RichardPerdaan — APPROVED** [review]

**Review by joerivanveen — APPROVED** [review]

**Review by markernst97 — APPROVED** [review]

**Comment by EdieLemoine** [comment]
This will (likely) be tested next week. When testing is approved, this PR will be merged and included in the next release. Thanks for contributing!

File: Model/Source/SourceItem.php [hunk 5]
```diff
@@ -13,31 +14,54 @@ class SourceItem
...
-            ->addFilter(SourceItemInterface::SKU, $sku)
-            ->create();
+        if ($this->sourceItemRepository) {
+            $searchCriteria = $this->searchCriteriaBuilder
+                ->addFilter(SourceItemInterface::SKU, $sku)
+                ->create();
+
+            return $this->sourceItemRepository->getList($searchCriteria)->getItems();
+        } 
+        return [];
```
Comment (RichardPerdaan) [inline Model/Source/SourceItem.php:51]:
```suggestion
        return [];
```

**Comment by RichardPerdaan** [comment]
Since version [4.3.0](https://github.com/myparcelnl/magento/releases/tag/4.3.0) we have made an adjustment, that solves also this problem. If it doesn't work, we can always reopen this PR

**Pull request closed without merging**